src/Microsoft.Identity.Client/TokenCache.cs

```diff
@@ -185,6 +185,8 @@ internal void OnBeforeWrite(TokenCacheNotificationArgs args)
 
             IdToken idToken = IdToken.Parse(response.IdToken);
 
+            string subject = idToken?.Subject;
```

Comment:
>string subject = idToken?.Subject; [](start = 12, length = 34)

what is the subject? What does ADFS send in here?

Reply:
The subject is the adfs equivalent of a unique identifier. This should always be returned in the ID token

Reply:
Should we throw an exception if it's not set? Or at least add this information to the log.

---
In reply to: [252027606](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/834#discussion_r252027606) [](ancestors = 252027606)